server/lib/models/channel.go

```diff
@@ -53,8 +78,30 @@ func ListChannel(dbif Dbif, req *proto.ChannelListRequest) ([]*proto.Model_Chann
 	} else {
 		limit = CHANNEL_FETCH_LIMIT
 	}
-	//TODO: respect category parameter of req.
-	chs, err := dbif.Select(Channel{}, dbm.Stmt("select * from %s.channels order by id desc limit $1"), limit)
+	tmp := make([]string, 0)
+	cond := ""
+	order_by := "id desc"
+	if req.Query == proto.ChannelListRequest_New {
```

Comment:
__[gometalinter]__
* [gotype] invalid operation: req (variable of type *invalid type) has no field or method Query
* [gotype] undeclared name: proto